packages/plugin-session-replay-browser/test/session-replay.test.ts

```diff
@@ -247,27 +246,82 @@ describe('SessionReplayPlugin', () => {
   });
 
   describe('recordEvents', () => {
+    test('should send the first two emitted events immediately', () => {
+      const sessionReplay = sessionReplayPlugin();
+      const send = jest.spyOn(sessionReplay, 'send').mockReturnValueOnce(Promise.resolve());
+      sessionReplay.config = mockConfig;
+      sessionReplay.recordEvents();
+      expect(sessionReplay.events).toEqual([]);
+      const recordArg = record.mock.calls[0][0];
+      recordArg?.emit && recordArg?.emit(mockEvent);
+      recordArg?.emit && recordArg?.emit(mockEvent);
```

Comment:
Woah - i'm unfamiliar with this library, can you help me understand what lines 255-257 are doing?

Reply:
I needed a way to test the internal functionality of the callback that we pass to rrweb here: https://github.com/amplitude/Amplitude-TypeScript/blob/721a3997673d700b6bda9302b076be0f3fab7c09/packages/plugin-session-replay-browser/src/session-replay.ts#L93-L93
So all this is doing is grabbing the callback from the argument passed to the rrweb record (which is spied on as `record` here) and manually calling it with a mock event, like rrweb itself will do when performing recordings